armory/scenarios/ucf101.py

```diff
@@ -0,0 +1,197 @@
+"""
+Classifier evaluation within ARMORY
+
+Contribuor: MITRE Corporation
+"""
+
+import importlib
+import logging
+import time
+
+import numpy as np
+from tqdm import tqdm
+
+from armory.scenarios.base import Scenario
+from armory.utils.metrics import AverageMeter
+from armory.utils.config_loading import load_dataset, load_model
+
+logger = logging.getLogger(__name__)
+
+
+class Ucf101(Scenario):
+    def _evaluate(self, config: dict) -> dict:
+        """
+        Evaluate the config and return a results dict
+        """
+        model_config = config["model"]
+        classifier, preprocessing_fn = load_model(model_config)
+
+        if model_config.get("fit"):
+            logger.info(
+                f"Fitting model of {model_config['module']}.{model_config['name']}..."
+            )
+            logger.info(f"Loading training dataset {config['dataset']['name']}...")
+            train_epochs = config["model"]["fit_kwargs"]["nb_epochs"]
+            batch_size = config["dataset"]["batch_size"]
+            train_data_generator = load_dataset(
+                config["dataset"],
+                epochs=train_epochs,
+                split_type="train",
+                preprocessing_fn=preprocessing_fn,
+            )
+
+            for epoch in range(train_epochs):
+                classifier.set_learning_phase(True)
+                start = time.time()
+
+                for _ in tqdm(
+                    range(train_data_generator.batches_per_epoch),
+                    desc=f"Epoch: {epoch}/{train_epochs}",
+                ):
+                    x_trains, y_trains = train_data_generator.get_batch()
+                    # x_trains consists of one or more videos, each represented as an
+                    # ndarray of shape (n_stacks, 3, 16, 112, 112).
+                    # To train, randomly sample a batch of stacks
+                    x_train = np.zeros(
+                        (min(batch_size, len(x_trains)), 3, 16, 112, 112),
+                        dtype=np.float32,
+                    )
+                    for i, xt in enumerate(x_trains):
+                        rand_stack = np.random.randint(0, xt.shape[0])
+                        x_train[i, ...] = xt[rand_stack, ...]
+                    classifier.fit(
+                        x_train, y_trains, batch_size=batch_size, nb_epochs=1
+                    )
+                logger.info(f"Epoch {epoch}/{train_epochs} took {time.time() - start}s")
+
+                # evaluate on test examples
+                classifier.set_learning_phase(False)
```

Comment:
There are a number of models that can do things differently during prediction. For PyTorch, I think this basically sets `torch.no_grad()` so that it doesn't keep track of gradients while computing.